src/pkg/packager/deploy.go

```diff
@@ -459,10 +460,29 @@ func (p *Packager) installChartAndManifests(componentPath types.ComponentPaths,
 	}
 
 	for _, manifest := range component.Manifests {
-		for idx := range manifest.Kustomizations {
-			// Move kustomizations to files now
-			destination := fmt.Sprintf("kustomization-%s-%d.yaml", manifest.Name, idx)
-			manifest.Files = append(manifest.Files, destination)
+		for idx, k := range manifest.Kustomizations {
+			// Generate manifests from kustomizations and place in the package
+			builtFile := fmt.Sprintf("kustomization-%s-%d.yaml", manifest.Name, idx)
+			destination := filepath.Join(componentPath.Manifests, builtFile)
+
+			// Check if the destination exists, if it does, this means that this package is v0.22.2 or older and already has built kustomizations
+			if utils.InvalidPath(destination) {
+				kPath := filepath.Join(componentPath.Manifests, k)
+
+				// Run the template engine against the kustomizations (if they exist) before they are built
+				valueTemplate, err := template.Generate(p.cfg)
+				if err != nil {
+					return installedCharts, err
+				}
+				template.ProcessYamlFilesInPath(kPath, component, valueTemplate)
+
+				if err := kustomize.BuildKustomization(kPath, destination, manifest.KustomizeAllowAnyDirectory); err != nil {
```

Comment:
I'm worried about this and remote bases in the airgap.  Does the Kustomize build properly replace a kustomize file like this one:

https://github.com/defenseunicorns/zarf/blob/main/packages/big-bang-core/kustomization/core-standard/kustomization.yaml#L2

It seems like it now may require connectivity for the kustomize build to access that address.

Reply:
We may need to run something like this when it is available or break the build into two steps: https://github.com/kubernetes-sigs/kustomize/issues/3980